src/components/search/GlobalSearchField.js

```diff
@@ -429,6 +436,14 @@ function GlobalSearchField(props) {
         setFilter(selectedFilter);
     }, [selectedFilter]);
 
+    useEffect(() => {
+        setDefaultTab(
+            config?.elasticEnabled
+                ? SEARCH_RESULTS_TABS.data
+                : SEARCH_RESULTS_TABS.apps
+        );
+    }, [config]);
```

Comment:
I don't think this effect and the `defaultTab` state is required.
It should be enough to replace the state and this effect with a `const`:
```jsx
const defaultTab = config?.elasticEnabled
    ? SEARCH_RESULTS_TABS.data
    : SEARCH_RESULTS_TABS.apps;
```
Since this component is calling the `useConfig` hook, then it should re-render with any `config` updates, which should also update this `defaultTab` const.

For reference, see https://react.dev/learn/you-might-not-need-an-effect

Reply:
Thank you for this reference. I was debating whether this should be an effect and this guide will be helpful.